src/dune_engine/fs_cache.mli

```diff
@@ -70,8 +56,6 @@ end
 module Untracked : sig
   val path_stat : (Reduced_stats.t, Unix_error.Detailed.t) result t
 
-  val file_digest : Cached_digest.Digest_result.t t
```

Comment:
I'm a little sad that the implementation is now not uniform, i.e. some file-system caches are defined in this module and some in `cached_digest`. I don't know if it's easy to bring back the uniformity but I think we should at least add a comment above saying that file digests are handled specially.

Reply:
Well, we can keep the interface uniform if you'd like. But I was thinking that this code needs a little more refactoring before we decide on good interfaces. One thing I don't quite like and I was going to bring up is how caching is coupled with saving/restoring from disk. For example:

* it would be nice to restore/save digests and stats from only the build directory. I don't think that's possible with the single table.
* it would be nice to unify the stat tables. But using the stat table in Cached_digest calculates the digests eagerly.

Reply:
OK, I agree. I think it's worth writing your comment in the code somewhere.